api/db/changelog/db.changelog-57-add-era-commons-completion-time.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9
+                    http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+  <changeSet author="blrubenst" id="db.changelog-57-add-era-commons-completion-time">
+    <addColumn tableName="user">
+      <column name="era_linked_nih_username" type="varchar(255)"/>
+      <column name="era_link_expire_time" type="datetime"/>
```

Comment:
I think we're missing a third column here: the core "moduleCompletionTime" column, which tracks the last timestamp at which a given access module was known to be completed by a user.

This column would be named like

era_commons_completion_time

or

era_link_completion_time

depending on which prefix we choose (see above comment).

This column would capture the timestamp when the user goes from an unlinked to linked state. And would be cleared when going from a linked to unlinked state. This and the bypass field would be guaranteed to be present for all access modules, and this is what we'd be looking at from L105 of UserService.java.

The motivation here was that by ensuring all access modules are modeled with this bare minimum of similarly-typed fields, it becomes much clearer to treat these different modules in a consistent way. (Versus the alternative, where one module has to be checked against an empty string, another module is a boolean, another is an enum, etc... which quickly becomes unnecessary mental burden to grok)